packages/lib/src/core/core.component.props.test.ts

```diff
@@ -0,0 +1,306 @@
+            expect(component.props.clientKey).toEqual('test_F7_FEKJHF');
+            expect(component.props.amount.value).toEqual(19000);
+
+            // expect props from relevant pmResponse paymentMethod
+            expect(component.props.configuration.gatewayMerchantId).toEqual('TestMerchantCheckout');
+
+            // expect props from relevant paymentMethodsConfiguration object
+            expect(component.props['foo']).toEqual('bar');
+
+            // expect props from GooglePay.formatProps()
+            expect(component.props.type).toEqual('paywithgoogle');
+            expect(component.props.buttonSizeMode).toEqual('static');
+        });
+    });
+
+    /**
+     * DROPIN
+     */
+    describe('Test for dropin component', () => {
+        test('Dropin component receives correct props ', () => {
+            const checkout = new AdyenCheckout(checkoutConfig);
+            const dropin = checkout.create('dropin', {
+                showStoredPaymentMethods: false,
+                openFirstPaymentMethod: false
+            });
+
+            // expect props from core.getPropsForComps()
+            expect(dropin.props._parentInstance).not.toEqual(null);
+            expect(dropin.props.paymentMethods).toEqual(paymentMethodsResponse.paymentMethods);
+
+            // expect props from core.processGlobalOptions
+            expect(dropin.props.clientKey).toEqual('test_F7_FEKJHF');
+            expect(dropin.props.amount.value).toEqual(19000);
+
+            // expect props from config object passed when dropin is created
+            // expect(dropin.props.showStoredPaymentMethods).toEqual(false);
+            expect(dropin.props.openFirstPaymentMethod).toEqual(false);
+
+            //
+            expect(dropin.props.type).toEqual('dropin');
+        });
+    });
+
+    /**
+     * Cannot get this test to work - I think the async nature of how the Dropin is created means the dropin.dropinRef.state object is not
+     * populated on time (the activePaymentMethod stays as null & the elements array stays empty), so we cannot gain the access to the lower level
+     * props that we need for this test to work.
+     * Leaving it here in case someone know how to get it to works - in the meantime switching to an e2e test for this: dropin.components.props.test.js
+     */
+    describe.skip('Tests for dropin created components', () => {
+        let dropin;
+        // Need to instantiate this way for the async mounting to work
+        beforeEach(() => {
+            const checkout = new AdyenCheckout({});
+            dropin = checkout.create('dropin');
+        });
+
+        test('StoredCard in Dropin receives correct props ', async () => {
+            const wrapper = await mount(dropin.render());
+            wrapper.update();
+
+            console.log('### core.component.props.test::dropin.dropinRef:: ', dropin.dropinRef.state); // Never fully populated
+
+            const storedCard = dropin.dropinRef.state.elements[0];
```

Comment:
To make this test work, first you need to pass the config to AdyenCheckout in the beforeEach method:
```javascript
const checkout = new AdyenCheckout(checkoutConfig);
```

Secondly, after mounting the component, you need to flush all the promises. There are pending promises from the `createElements.ts` file which are doing some async task to check if payment method `isAvailable`.

```javascript
test('StoredCard in Dropin receives correct props ', async () => {
    mount(dropin.render());
    const flushPromises = () => new Promise(resolve => setImmediate(resolve));
    await flushPromises();
    ...
```

And the third change, is to remove `GooglePay` from the `paymentMethods` object for this test scenario, as GooglePay does some different async request on `isAvailable` to check if it can be rendered or not. I believe you can also mock it if you do not want to remove it, that is up to you.

Reply:
@ribeiroguilherme - thanks for the tips. I'll look into implementing them!

Reply:
Have got these unit tests running thanks @ribeiroguilherme